src/ShellCheck/Checks/Commands.hs

```diff
@@ -620,6 +622,9 @@ checkInteractiveSu = CommandCheck (Basename "su") f
     undirected (T_Redirecting _ (_:_) _) = False
     undirected _ = True
 
+    noCommandArgPresent = not . any isCommandArg . map (getLiteralStringDef "_")
```

Comment:
There are some other valid ways of specifying a command, such as `su root ./myscript`, while noobs more frequently just use `su`, `su -`, or `su root` when encountering this pitfall